src/command_modules/azure-cli-servicefabric/azure/cli/command_modules/servicefabric/tests/test_sf_commands.py

```diff
@@ -0,0 +1,87 @@
+
+    def body(self):
+        rg = self.resource_group
+        name = self.cluster_name
+        location = self.location
+
+        self.cmd('sf cluster list', checks=[JMESPathCheck('type(@)', 'array')])
+
+        self.cmd('sf cluster show --resource-group {} --name {}'.format(rg,name), 
+                 checks=[JMESPathCheck('name', name)])
+
+        self.cmd('sf cluster upgrade-type set --resource-group {} --cluster-name {} --upgrade-mode automatic'.format(rg,name), 
+                 checks=[JMESPathCheck('upgradeMode', 'Automatic')])
+
+        test_thumbprint = '9B609A389BD4597BEEFABFB363BF2BBF2E806001'
+        self.cmd('sf cluster client-certificate add --resource-group {} --name {}  --thumbprint {}'.format(rg,name,test_thumbprint), 
+                 checks=[JMESPathCheck('clientCertificateThumbprints[0].certificateThumbprint', test_thumbprint),
+                     JMESPathCheck('clientCertificateThumbprints[0].isAdmin', False)])
+
+        self.cmd('sf cluster client-certificate remove --resource-group {} --name {} --thumbprints {}'.format(rg,name,test_thumbprint), 
+                 checks=[JMESPathCheck('length(clientCertificateThumbprints)', 0)])
+
+        cluster = self.cmd('sf cluster certificate add --resource-group {} --cluster-name {} --certificate-subject-name test.com'.format(rg,name))
+        assert cluster['certificate']['thumbprintSecondary']
+
+        cluster = self.cmd('sf cluster certificate remove --resource-group {} --cluster-name {} --thumbprint {}'.format(rg,name,cluster['certificate']['thumbprintSecondary']))
+        assert not cluster['certificate']['thumbprintSecondary']
+
+        self.cmd('sf cluster setting set --resource-group {} --name {} --section NamingService --parameter MaxOperationTimeout --value 10000'.format(rg,name),
+                           checks=[JMESPathCheck('fabricSettings[1][name]', 'NamingService'),
+                               JMESPathCheck('fabricSettings[1][parameters][name]', 'MaxOperationTimeout'),
+                               JMESPathCheck('fabricSettings[1][parameters][value]', '10000')])
+       
+        self.cmd('sf cluster setting remove --resource-group {} --name {} --section NamingService --parameter MaxOperationTimeout'.format(rg,name),
+                           checks=[JMESPathCheck('length(fabricSettings)', 0)])
+
+        self.cmd(' sf cluster reliability update --resource-group {} --name {} --reliability-level Bronze'.format(rg,name),
+                           checks=[JMESPathCheck('reliabilityLevel', 'Bronze')])
+
+        self.cmd(' sf cluster reliability update --resource-group {} --name {} --durability-level Bronze'.format(rg,name),
+                           checks=[JMESPathCheck('nodeTypes[0].durabilityLevel', 'Bronze')])
+
+        self.cmd('sf application certificate add --resource-group {} --cluster-name {} --certificate-subject-name test2.com'.format(rg,name),
+                           checks=NoneCheck())
+
+if __name__ == '__main__':
```

Comment:
I might have missed some, but it looks like tests are missing for the following commands:
`sf cluster setting remove`, `sf cluster durability update`, `sf cluster node-type add`, `sf cluster node add/remove`